File: DomainSpecificLanguages/DomainSpecificLanguages/AlarmsWithNestedFunctions/SwitchCriterias/CompoundSwitchCriteria.cs
using System.Linq;

namespace AlarmsWithNestedFunctions.SwitchCriterias
{
  public class CompoundSwitchCriteria : SwitchCriteria
  {
    private readonly SwitchCriteria[] _switchCriterias;

    public CompoundSwitchCriteria(params SwitchCriteria[] switchCriterias)
    {
      _switchCriterias = switchCriterias;
    }

    public bool IsNotMet()
    {
      return _switchCriterias.All(criteria => criteria.IsNotMet());
    }
  }
}

File: Java/dependency-injection/changing-behavior-through-composition/src/main/java/com/github/grzesiek_galezowski/ChangingBehaviorThroughComposition/SwitchCriterias/CompoundSwitchCriteria.java
package com.github.grzesiek_galezowski.ChangingBehaviorThroughComposition.SwitchCriterias;

import com.github.grzesiek_galezowski.ChangingBehaviorThroughComposition.SwitchCriteria;

import java.util.Arrays;

public class CompoundSwitchCriteria implements SwitchCriteria
  {
    private final SwitchCriteria[] _switchCriterias;

    public CompoundSwitchCriteria(SwitchCriteria...switchCriterias)
    {
      _switchCriterias = switchCriterias;
    }

    public boolean isNotMet()
    {
      return Arrays.stream(_switchCriterias).anyMatch(criteria -> criteria.isNotMet());
    }
  }
